numpy/lib/tests/test_nanfunctions.py

```diff
@@ -588,6 +588,15 @@ def test_empty(self):
                     assert_(len(w) == 0)
 
 
+_TIME_UNITS = (
+    "Y", "M", "W", "D", "h", "m", "s", "ms", "us", "ns", "ps", "fs", "as"
+)
+
+# All `inexact` + `timdelta64` type codes
+_TYPE_CODES = list(np.typecodes["AllFloat"])
+_TYPE_CODES += [f"m8[{unit}]" for unit in _TIME_UNITS]
+
```

Comment:
Would it make sense to include `datetime64` as well?

Reply:
For `np.nanmedian`? No, it wouldn't.
More generally, `np.median` uses `np.add.reduce` under the hood, which lacks the necessary support for a `MM->M`-type signatures.